models/user/badge.go

```diff
@@ -122,3 +226,55 @@ func RemoveAllUserBadges(ctx context.Context, u *User) error {
 	_, err := db.GetEngine(ctx).Where("user_id=?", u.ID).Delete(&UserBadge{})
 	return err
 }
+
+// SearchBadgeOptions represents the options when fdin badges
+type SearchBadgeOptions struct {
+	db.ListOptions
+
+	Keyword string
+	Slug    string
+	ID      int64
+	OrderBy db.SearchOrderBy
+	Actor   *User // The user doing the search
+}
+
+func (opts *SearchBadgeOptions) ToConds() builder.Cond {
+	cond := builder.NewCond()
+
+	if opts.Keyword != "" {
+		lowerKeyword := strings.ToLower(opts.Keyword)
```

Comment:
Why lower case is necessary?

Reply:
copied from other keyword search code for repos and users, but I kept it since the slug's are kept as lowercase already. I can remove it if you think it's not needed.